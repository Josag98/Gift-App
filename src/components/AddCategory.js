import React,{ useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({setCategories}) => {

    const [inputValue, setinputValue] = useState('')

    const handleOnChange =(e)=>{
        setinputValue(e.target.value)
    }

    const handleSubmit = (e) =>
    {
        e.preventDefault();

        if(inputValue.trim().length>2){
            setCategories(cat=>[inputValue,...cat]);
            setinputValue('')
        }
        

    }
    

  return (
      <form onSubmit={handleSubmit}>
    <input
    placeholder='Â¿Que gift desea buscar?'
    type="text"
    value={inputValue}
    onChange={handleOnChange}
    >
        
    </input>
    </form>
  )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
