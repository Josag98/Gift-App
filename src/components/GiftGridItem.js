import React from 'react'
import PropTypes from "prop-types";

export const GiftGridItem =({title,url}) => {
  
  return (
    <div className='card animate__animated animate__swing'>
        <img src={url} alt={title}></img>
        <p>{title}</p>
    </div>
  )
}

GiftGridItem.propTypes={
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}
