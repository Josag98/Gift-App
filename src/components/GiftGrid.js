import React, { useEffect,useState } from 'react'
import { getGifts } from '../helpers/getGifts';
import {GiftGridItem} from './GiftGridItem';

export const GiftGrid = ({category}) => {
  const [images, setImages] = useState([]);

  

  useEffect(() => {
    getGifts(category).then(imgs => setImages(imgs))
  }, [category]);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((img) => {
          return <GiftGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
}
