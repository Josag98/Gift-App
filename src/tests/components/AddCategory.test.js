import React from 'react';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';


describe('Pruebas del componente AddCategory.js', () => { 

    const setCategories=()=>{}
    const wrapper = shallow(<AddCategory setCategories={setCategories}/>)

    test('Se debe de renderizar el componente correctamente ', () => { 

       

        expect(wrapper).toMatchSnapshot();
     })

     test('Debe de cambiar la caja de texto', () => { 

        const input = wrapper.find('input');
        const value = 'Hola Mundo';

        input.simulate('change', {target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
      })
 })