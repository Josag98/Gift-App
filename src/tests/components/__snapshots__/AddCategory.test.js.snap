// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente AddCategory.js Se debe de renderizar el componente correctamente  1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    placeholder="Â¿Que gift desea buscar?"
    type="text"
    value=""
  />
</form>
`;
